### PR TITLE
Addresses #25 - API Endpoint for Detailed venue information

### LINKED ISSUE #25: [Backlog] Implement API-Endpoint for more detailed venue information
Story points: 5

Acceptance Criteria:
- [x] The API Endpoint should be usable by the frontend
- [x] The API Endpoint should have the unique identifier of a venue as the input argument
- [ ] The API Endpoint should return the following data:
    - [x] ID of the venue
    - [x] Name of the venue
    - [x] Longitude of the venue
    - [x] Latitude of the venue
    - [x] Address of the venue, in human-readable text, e.g. US format
    - [x] Categories of the venue
    - [x] URL of the Venue
    - [x] Opening Hours
    - [x] Popular Hours
    - [x] Price class of the venue (from 1 to 4)
    - [x] Rating of the venue
    - [x] Number of people now at the venue
    - [x] Number of likes

User Stories:
- As a frontend developer, I would like to have an API-Endpoint in the backend for detailed venue information, such that I can use this for providing dynamic content on the activity page and the map.
- As a trip planner, I would like to see more specific venue information, such that I know better what to expect from each activity or venue, like if a certain activity will cost a lot or not.

### PR DESCRIPTION
**Pull request by Producermark**
Closes #25

### CODE REVIEW

File: backend/app/ContentProviders/FourSquare.php [hunk 1]
```diff
@@ -67,11 +69,15 @@ public function get(String $endpoint, String $method, array $parameters)
...
-        $venue->setAddressHumanReadable($object->location->formattedAddress);
-        $venue->setLatitude($object->location->lat);
-        $venue->setLongitude($object->location->lng);
+        $this->setBasicVenueItems($object, $venue);
+        return $venue;
+    }
+
+    public function createDetailedVenueFromAPIListing($object): Venue
+    {
+        $venue = new DetailedVenue();
```
Comment (killje) [inline backend/app/ContentProviders/FourSquare.php]: Can you not give the object to the venue and have the venue set the items and remove all the setter functions from the class. This way all the venue logic is inside tge venue

Comment (Producermark) [reply backend/app/ContentProviders/FourSquare.php]: I chose to do it this way, such that there is a class FourSquare which will handle all FourSquare's logic. The rest of the application is in no way dependent on the FourSquare API, so if we would like to use another API instead, only the methods in FourSquare.php will need to be copied and modified for that API.
TL;DR it's because of more modularity

Comment (killje) [reply backend/app/ContentProviders/FourSquare.php]: Then i would opt for a abstract class for the data and four square class that extend it.

Comment (Producermark) [reply backend/app/ContentProviders/FourSquare.php]: There already is. The `ContentProvider` class, which FourSquare extends.

Comment (killje) [reply backend/app/ContentProviders/FourSquare.php]: Since it is not important for the API right now that this changes i will accept it, but would still like to see this changed in a new pull request
